app/(tabs)/profile.tsx

```diff
@@ -41,14 +43,13 @@ export default function Profile() {
         const body: UserResponseBodyGet = await response.json();
 
         if ('error' in body) {
-          Alert.alert('Error', body.error, [{ text: 'OK' }]);
-          return router.push('/(auth)/login');
+          router.replace('/(auth)/login?returnTo=/(tabs)/profile');
```

Comment:
let's match the pattern from above, even though we don't need the `return`

```suggestion
          router.replace('/(auth)/login?returnTo=/(tabs)/profile');
          return;
```